stl/inc/ranges

```diff
@@ -2635,6 +2589,415 @@ namespace ranges {
         inline constexpr _Drop_while_fn drop_while;
     } // namespace views
 
+    // CLASS TEMPLATE ranges::join_view
+    // clang-format off
+    template <input_range _Vw>
+        requires view<_Vw> && input_range<range_reference_t<_Vw>>
+            && (is_reference_v<range_reference_t<_Vw>> || view<range_value_t<_Vw>>)
+    class join_view;
+    // clang-format on
+
+    template <class _Vw>
+    class _Join_view_base : public view_interface<join_view<_Vw>> {
+    protected:
+        /* [[no_unique_address]] */ views::all_t<range_reference_t<_Vw>> _Inner{};
+    };
+
+    // clang-format off
+    template <class _Vw>
+        requires is_reference_v<range_reference_t<_Vw>>
+    class _Join_view_base<_Vw> : public view_interface<join_view<_Vw>> {};
+
+    template <input_range _Vw>
+        requires view<_Vw>&& input_range<range_reference_t<_Vw>>
+            && (is_reference_v<range_reference_t<_Vw>> || view<range_value_t<_Vw>>)
+    class join_view : public _Join_view_base<_Vw> {
+        // clang-format on
+    private:
+        using _InnerRng = range_reference_t<_Vw>;
+        /* [[no_unique_address]] */ _Vw _Range{};
+
+        template <bool _Const>
+        class _Sentinel;
+
+        template <class _OuterTraits, class _InnerTraits, bool _Deref_is_glvalue> // TRANSITION, LWG-3289
+        struct _Category_base {};
+
+        // clang-format off
+        template <class _OuterTraits, class _InnerTraits, bool _Deref_is_glvalue>
+            requires _Has_member_iterator_category<_InnerTraits> && _Has_member_iterator_category<_OuterTraits>

```

Comment:
Nitpick: Can the `requires` mention its conditions in the same order as the template parameters?